static/app/views/replays/detail/memoryPanel/domNodesChart.tsx

```diff
@@ -0,0 +1,184 @@
+        const firstValue = Array.isArray(values) ? values[0] : values;
+        const seriesTooltips = toArray(values).map(
+          value => `
+            <div>
+              <span className="tooltip-label">${value.marker}<strong>${value.seriesName}</strong></span>
+              ${value.data[1]}
+            </div>
+          `
+        );
+
+        return `
+          <div class="tooltip-series">${seriesTooltips.join('')}</div>
+            <div class="tooltip-footer">
+              ${t('Date: %s', getFormattedDate(startOffsetMs + firstValue.axisValue, 'MMM D, YYYY hh:mm:ss A z', {local: false}))}
+            </div>
+            <div class="tooltip-footer" style="border: none;">
+              ${t('Time within replay: %s', formatTime(firstValue.axisValue))}
+            </div>
+          <div class="tooltip-arrow"></div>
+        `;
+      },
+    }),
+    xAxis: XAxis({
+      type: 'time',
+      axisLabel: {
+        formatter: (time: number) => formatTime(time, false),
+      },
+      theme,
+    }),
+    yAxis: YAxis({
+      type: 'value',
+      name: t('DOM Nodes'),
+      theme,
+      nameTextStyle: {
+        padding: [8, 8, 8, 48],
+        fontSize: theme.fontSizeLarge,
+        fontWeight: 600,
+        lineHeight: 1.2,
+        fontFamily: theme.text.family,
+        color: theme.gray300,
+      },
+      minInterval: 100,
+      maxInterval: Math.pow(1024, 4),
+      axisLabel: {
+        formatter: (value: number) => axisLabelFormatter(value, 'number', true),
+      },
+    }),
+    onMouseOver: ({data}) => {
+      if (data[0]) {
+        setCurrentHoverTime(data[0]);
+      }
+    },
+    onMouseOut: () => {
+      setCurrentHoverTime(undefined);
+    },
+    onClick: ({data}) => {
+      if (data.value) {
+        setCurrentTime(data.value);
+      }
+    },
+  };
+
+  const staticSeries: Series[] = useMemo(
+    () => [
+      {
+        id: 'nodeCount',
+        seriesName: t('Number of DOM nodes'),
+        data: datapoints.map(d => ({
+          value: d.count,
+          name: d.endTimestampMs - startTimestampMs,
+        })),
+        stack: 'dom-nodes',
+        lineStyle: {opacity: 0, width: 2},
+      },
+      {
+        id: 'replayStart',
+        seriesName: 'Replay Start',
+        data: [{value: 0, name: 0}],
+        lineStyle: {opacity: 0, width: 0},
+      },
+      {
+        id: 'replayEnd',
+        seriesName: 'Replay End',
+        data: [{value: 0, name: durationMs}],
+        lineStyle: {opacity: 0, width: 0},
+      },
+    ],
+    [datapoints, durationMs, startTimestampMs]
+  );
+
+  const currentTimeSeries = useMemo(
+    (): Series => ({
+      id: 'currentTime',
+      seriesName: t('Current player time'),
+      data: [],
+      markLine: {
+        symbol: ['', ''],
+        data: [{xAxis: currentTime}],
+        label: {show: false},
+        lineStyle: {type: 'solid', color: theme.purple300, width: 2},
+      },
+    }),
+
+    [currentTime, theme.purple300]
```

Comment:
Is `useMemo` the right thing to use here since currentTime is alway changing (assuming replay is playing).

Reply:
if the replay is paused then that timestamp won't change